### PR TITLE
Cache `isModuleOutputEnabled` or `isModuleEnabled` status

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

If `isModuleOutputEnabled` or `isModuleEnabled` is called multiple times for the same Module, get config once.

### Related Pull Requests

1. see OpenMage/magento-lts#4320

### Manual testing scenarios (*)

1. add xdebug breapoints to `isModuleOutputEnabled()`
2. ...

```php
    var_dump($test->isModuleOutputEnabled('Mage_Catalog'));
    var_dump($test->isModuleOutputEnabled('Mage_Catalog'));
    var_dump($test->isModuleOutputEnabled('Mage_Catalog'));
    var_dump($test->isModuleOutputEnabled('a'));
    var_dump($test->isModuleOutputEnabled('a'));
    var_dump($test->isModuleOutputEnabled('b'));
```

### CODE REVIEW

File: app/code/core/Mage/Core/Helper/Abstract.php [hunk 1]
```diff
@@ -42,6 +42,10 @@ abstract class Mage_Core_Helper_Abstract
      */
     protected $_layout;
 
+    protected array $modulesDisabled = [];
```
Comment (kiatng) [inline app/code/core/Mage/Core/Helper/Abstract.php:45]: Traditionally, protected var has prefix '_'. Do we want to break tradition?

Comment (sreichel) [reply app/code/core/Mage/Core/Helper/Abstract.php:45]: i'd not stick with old traditions.

i think we had PRs that added protected properties w/o underscore.

**Comment by justinbeaty** [comment]
XML / DB config is already cached, how much time is this actually saving?

**Comment by sreichel** [comment]
xh1

For the example above,

**Comment by sreichel** [comment]
> even though the "Disable Modules Output" can be configured at the store scope

I did not see that and did not test it. Thanks.

Config is alerady cached, so it brings no benfits. Check if module is enabled still save some calls. (seems even faster ... numbers from ddev + xhprof + xhgui)

**Review by justinbeaty — APPROVED** [review]

**Comment by sreichel** [comment]
https://github.com/OpenMage/magento-lts/pull/4320#discussion_r1821212246
> However, I only see two calls to this method

Updated:

- moved methods to parent abstract class ... so its also available for frontend block classes too
- added trait to provide same funtionality to models, resource,-models, ... (instead of adding helper methods multiple times)

**Review by kiatng — DISMISSED** [review]

**Review by kiatng — APPROVED** [review]

**Review by addison74 — APPROVED** [review]

**Pull request merged**